app/uk/gov/hmrc/helptosavereminder/actors/ProcessingSupervisor.scala

```diff
@@ -55,7 +66,9 @@ class ProcessingSupervisor @Inject()(
   lazy val userScheduleCronExpression
     : String = config.getOptional[String](s"userScheduleCronExpression") map (_.replaceAll("_", " ")) getOrElse ("")
 
-  lazy val repoLockPeriod: Int = config.getOptional[Int](s"mongodb.repoLockPeriod").getOrElse(55)
+  val defaultRepoLockPeriod: Int = 55
```

Comment:
This should be in AppConfig

Reply:
The whole idea is to provide a default value in case the configured value 'mongodb.repoLockPeriod' is missing. Its used inside the getOrElse part of getOptional. I think I should straight away use 55 in getOrElse part and suppress the magic number warning using //scalastyle:ignore magic.number instead of declaring a immutable defaultRepoLockPeriod ?

Reply:
Yes, I realise, I'm not asking for a change to the magic number or to the getOrElse.  But my point is that the reading and validation of configuration should be done in the AppConfig object.

Apart from imposing a familiar program structure, the advantage is that configuration is more likely to be will be validated every time and at startup, rather than dynamically, the first time (if ever) that it is used.

Do you see?

Reply:
Oh, yeah, I declare this in AppConfig and read from there. Got it. Thanks.

Reply:
Likewise userScheduleCronExpression

Reply:
Sorry, Mohan, but you do not seem to understood what I have said: "Apart from imposing a familiar program structure, the advantage is that configuration is more likely to be will be validated every time and at startup, rather than dynamically, the first time (if ever) that it is used.", because you have not moved the calls to getOptional into AppConfig,

Reply:
Ok Dave. Then we should be able to access the created immutables straight like:
appConfig.defaultRepoLockPeriod

Reply:
Yes, that is right

Reply:
Done